### DISCUSSION TITLE
Steps to rename an addon?

### DISCUSSION THREAD

**Discussion by ssyberg** [aws/copilot-cli#5016] [General]
I know it's a little silly since the addon name doesn't really affect anything, but I'd really like to rename an environment addon and all references. This is easy enough to do with a search/replace, but I tried that and got a bunch of errors on deploy (not too surprising). I'm guessing the only real way to do this is to delete the existing addon and then add a new one?

**Comment by KollaAdithya** [comment]
Hello again @ssyberg 😄 

what do you mean by rename an addon?  Do you want to rename the cluster with in the addon CloudFormation template

**Reply by ssyberg** [reply]
I created a storage addon with the copilot cli and gave it a stupid name, so like:
```
copilot
-- enviornments
---- addons
------ poorly-named.yml
```

I'd like to rename "poorly-named.yml" to "really-good-name.yml" and also replace all references in the manifests like `{poorlynamedBucket.Arn}` to `{reallygoodnameBucket.Arn}`

**Reply by ssyberg** [reply]
I just don't totally understand what the implications are / how this is handled in the stored state

**Comment by KollaAdithya** [comment]
I just deployed an S3 bucket with  name `poorlynamed` and after that I replaced all the occurences of `poorlynamed`  with `goodnamed` bucket in the addon template. I was able to deployed the service with the addon with out any error.

I also renamed the file from `poorlynamed.yml` to `goodnamed.yml`

**Reply by ssyberg** [reply]
Awesome, I'll try it again
